[BUG] "isolated_limited_api_tests" crashes in Py3.13a4

### Describe the bug

The logs aren't really helpful (worth improving), they just show this:
```
======================================================================
FAIL: runTest (__main__.EndToEndTest.runTest)
[6] End-to-end isolated_limited_api_tests
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cython/cython/runtests.py", line 2028, in runTest
    self.assertEqual(0, res, "non-zero exit status, last output was:\n%r\n-- stdout:%s\n-- stderr:%s\n" % (
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ' '.join(command), out[-1], err[-1]))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 0 != -11 : non-zero exit status, last output was:
'/opt/hostedtoolcache/Python/3.13.0-alpha.4/x64/bin/python run.py'
-- stdout:
-- stderr:
```

### Code to reproduce the behaviour:

_No response_

### Expected behaviour

_No response_

### OS

_No response_

### Python version

3.13a4

### Cython version

_No response_

### Additional context

_No response_

```
limited.c: In function ‘PyInit_limited’:
limited.c:1030:45: warning: implicit declaration of function ‘PyImport_AddModuleRef’; did you mean ‘PyImport_AddModule’? [-Wimplicit-function-declaration]
 1030 |   #define __Pyx_PyImport_AddModuleRef(name) PyImport_AddModuleRef(name)
      |                                             ^~~~~~~~~~~~~~~~~~~~~
limited.c:6456:13: note: in expansion of macro ‘__Pyx_PyImport_AddModuleRef’
 6456 |   __pyx_b = __Pyx_PyImport_AddModuleRef(__Pyx_BUILTIN_MODULE_NAME); if (unlikely(!__pyx_b)) __PYX_ERR(0, 2, __pyx_L1_error)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
limited.c:6456:11: warning: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 6456 |   __pyx_b = __Pyx_PyImport_AddModuleRef(__Pyx_BUILTIN_MODULE_NAME); if (unlikely(!__pyx_b)) __PYX_ERR(0, 2, __pyx_L1_error)
      |           ^
limited.c:6457:24: warning: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 6457 |   __pyx_cython_runtime = __Pyx_PyImport_AddModuleRef("cython_runtime"); if (unlikely(!__pyx_cython_runtime)) __PYX_ERR(0, 2, __pyx_L1_error)
      |                        ^
limited.c: In function ‘__Pyx_FetchSharedCythonABIModule’:
limited.c:1030:45: warning: returning ‘int’ from a function with return type ‘PyObject *’ {aka ‘struct _object *’} makes pointer from integer without a cast [-Wint-conversion]
 1030 |   #define __Pyx_PyImport_AddModuleRef(name) PyImport_AddModuleRef(name)
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
limited.c:8910:12: note: in expansion of macro ‘__Pyx_PyImport_AddModuleRef’
 8910 |     return __Pyx_PyImport_AddModuleRef(__PYX_ABI_MODULE_NAME);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It's a 1-line fix - I'll push it shortly

--- a/Cython/Utility/ModuleSetupCode.c
+++ b/Cython/Utility/ModuleSetupCode.c
@@ -1244,7 +1244,7 @@ static CYTHON_INLINE PyObject * __Pyx_PyDict_GetItemStrWithError(PyObject *dict,
   #define __Pyx_PyUnicode_GET_LENGTH(o) PyUnicode_GetLength(o)
 #endif
 
-#if PY_VERSION_HEX >= 0x030d00A1
+#if __PYX_LIMITED_VERSION_HEX >= 0x030d00A1
   #define __Pyx_PyImport_AddModuleRef(name) PyImport_AddModuleRef(name)
 #else
   static CYTHON_INLINE PyObject *__Pyx_PyImport_AddModuleRef(const char *name) {

